packages/edge-config/src/utils/fetch-with-cached-response.ts

```diff
@@ -9,46 +11,115 @@ interface ResponseWithCachedResponse extends Response {
   cachedResponseBody?: unknown;
 }
 
-export const cache = new Map<string, CachedResponsePair>();
+/**
+ * A cache of request urls & auth headers and the resulting responses.
+ *
+ * This cache does not use Response instances as the cache value as reusing
+ * responses across requests leads to issues in Cloudflare Workers.
+ */
+export const cache = new Map<string, CachedResponseEntry>();
 
+/**
+ * Creates a new response based on a cache entry
+ */
+function createResponse(
+  cachedResponseEntry: CachedResponseEntry
+): ResponseWithCachedResponse {
+  return new Response(cachedResponseEntry.response, {
+    headers: cachedResponseEntry.headers,
+    status: cachedResponseEntry.status,
+  });
+}
+
+/**
+ * Used for bad responses like 500s
+ */
+function createHandleStaleIfError(cachedResponseEntry: CachedResponseEntry) {
+  return function handleStaleIfError(
+    response: ResponseWithCachedResponse
+  ): ResponseWithCachedResponse {
+    switch (response.status) {
+      case 500:
+      case 502:
+      case 503:
+      case 504:
+        return createResponse(cachedResponseEntry);
+      default:
+        return response;
+    }
+  };
+}
+
+/**
+ * Used on network errors which end up throwing
+ */
+function createHandleStaleIfErrorException(
+  cachedResponseEntry: CachedResponseEntry
+) {
+  return function handleStaleIfError(): ResponseWithCachedResponse {
+    return createResponse(cachedResponseEntry);
+  };
+}
+
+/**
+ * This is similar to fetch, but it also implements ETag semantics, and
+ * it implmenets stale-if-error semantics.
+ */
 export async function fetchWithCachedResponse(
   url: string,
   options: FetchOptions = {}
 ): Promise<ResponseWithCachedResponse> {
-  const { headers: customHeaders = new Headers(), ...customOptions } = options;
-  const authHeader = customHeaders.get('Authorization');
-  const cacheKey = `${url},${authHeader || ''}`;
+  try {
```

Comment:
I think this try-catch block should be removed.

It logs the error and then throws it, I think we only want to throw it and pass it on, to not pollute the output even if the error will be handled.

Reply:
🤦 Good catch! I added this for debugging purposes and forgot to remove it